pkg/audit/manager.go

```diff
@@ -403,13 +414,71 @@ func (am *Manager) auditResources(
 			}
 		}
 	}
+	// loop thru each subDir in output dir to get files
+	for i := 0; i < folderCount; i++ {
```

Comment:
Actually, it looks like we are auditing after we've looped through all GVKs, we'd definitely need to write out objects to separate directories by GVK in order to avoid clobbering.

One note about how we could be gentler on the API server...

If we perform the audit for a given kind before grabbing the next kind from the API server, we even out the load on the API server somewhat.

Reply:
Per above comment, I dont think we are clobbering it.

> If we perform the audit for a given kind before grabbing the next kind from the API server, we even out the load on the API server somewhat.

Can this optimization be a follow up PR given that we would need to do few rounds of load tests?

Reply:
Yeah, you're not clobbering it.

One other benefit of interleaving audits and lists... you can wipe the disk after each audit, lowering the maximum amount of disk space used.

Why would we need load tests?

In any case, we can defer, but the reduction in disk space seems valuable (akin to how much memory we got back when we switched to auditing on a per-kind basis)

Reply:
latest commit audits right after cache to disk for a given kind